File: src/main/java/com/alxnns1/mobhunter/blocks/MHPlant.java
package com.alxnns1.mobhunter.blocks;

import net.minecraft.block.BlockState;
import net.minecraft.block.BushBlock;
import net.minecraft.util.Direction;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockReader;
import net.minecraft.world.IWorldReader;

import javax.annotation.Nonnull;

public class MHPlant extends BushBlock {
	public MHPlant(Properties properties) {
		super(properties);
	}

	@Override
	public boolean isValidPosition(BlockState state, @Nonnull IWorldReader worldIn, BlockPos pos) {
		BlockPos blockpos = pos.down();
		return this.isValidGround(worldIn.getBlockState(blockpos), worldIn, blockpos);
	}

	@Override
	protected boolean isValidGround(BlockState state, @Nonnull IBlockReader worldIn, @Nonnull BlockPos pos) {
		return state.isSolidSide(worldIn, pos, Direction.UP);
	}
}

File: src/main/kotlin/com/alxnns1/mobhunter/block/MHPlant.kt
package com.alxnns1.mobhunter.block

import net.minecraft.block.BlockState
import net.minecraft.block.BushBlock
import net.minecraft.util.Direction
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockReader
import net.minecraft.world.IWorldReader

class MHPlant(val colour: Int, properties: Properties) : BushBlock(properties) {
	override fun isValidPosition(state: BlockState, world: IWorldReader, pos: BlockPos): Boolean {
		val posDown = pos.down()
		return isValidGround(world.getBlockState(posDown), world, posDown)
	}

	override fun isValidGround(state: BlockState, world: IBlockReader, pos: BlockPos): Boolean =
		state.isSolidSide(world, pos, Direction.UP)
}
